pkg/app/piped/notifier/slack.go

```diff
@@ -101,12 +104,19 @@ func (s *slack) sendEvent(ctx context.Context, event model.NotificationEvent) {
 		s.logger.Info(fmt.Sprintf("ignore event %s", event.Type.String()))
 		return
 	}
-	if err := s.sendMessage(ctx, msg); err != nil {
-		s.logger.Error(fmt.Sprintf("unable to send notification to slack: %v", err))
+	if len(s.config.HookURL) != 0 {
+		if err := s.sendMessageViaHookURL(ctx, msg); err != nil {
+			s.logger.Error(fmt.Sprintf("unable to send notification to slack: %v", err))
+		}
```

Comment:
Only one of `sendMessageViaHookURL` and `sendMessageViaAPI` should be called, so maybe we should add a return statement here to ensure we don't double send messages.